packages/salesforcedx-vscode-core/src/commands/forceOrgOpen.ts

```diff
@@ -6,36 +6,123 @@
   public build(data: {}): Command {
     return new SfdxCommandBuilder()
       .withDescription(nls.localize('force_org_open_default_scratch_org_text'))
       .withArg('force:org:open')
       .withLogName('force_org_open_default_scratch_org')
+      .withArg('--urlonly')
+      .withJson()
       .build();
   }
+
+  public buildUserMessageWith(orgData: OrgOpenSuccessResult): string {
+    return nls.localize(
+      'force_org_open_container_mode_message_text',
+      orgData.result.orgId,
+      orgData.result.username,
+      orgData.result.url
+    );
+  }
+
+  public execute(response: ContinueResponse<string>): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath(),
+      env: { SFDX_JSON_TO_STDOUT: 'true' }
+    }).execute(cancellationToken);
+
+    channelService.streamCommandStartStop(execution);
+
+    let stdOut: string = '';
+    execution.stdoutSubject.subscribe(cliResponse => {
+      stdOut += cliResponse.toString();
+    });
+
+    execution.processExitSubject.subscribe(() => {
+      this.logMetric(execution.command.logName, startTime);
+      try {
+        const orgOpenParser = new OrgOpenContainerResultParser(stdOut);
+
+        if (orgOpenParser.openIsSuccessful()) {
+          const cliOrgData = orgOpenParser.getResult() as OrgOpenSuccessResult;
+          const authenticatedOrgUrl: string = cliOrgData.result.url;
+
+          channelService.appendLine(this.buildUserMessageWith(cliOrgData));
+          // open the default browser
+          vscode.env.openExternal(vscode.Uri.parse(authenticatedOrgUrl));
+        } else {
+          const errorResponse = orgOpenParser.getResult() as OrgOpenErrorResult;
+          channelService.appendLine(errorResponse.message);
+        }
+      } catch (error) {
+        channelService.appendLine(
+          nls.localize('force_org_open_default_scratch_org_container_error')
+        );
+        telemetryService.sendError(
+          `forceOrgOpenContainer: There was an error when parsing the org open response ${error}`
+        );
+      }
+    });
+
+    notificationService.reportCommandExecutionStatus(
+      execution,
+      cancellationToken
+    );
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+export class ForceOrgOpenExecutor extends SfdxCommandletExecutor<{}> {
+  public build(data: {}): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_org_open_default_scratch_org_text'))
+      .withArg('force:org:open')
+      .withLogName('force_org_open_default_scratch_org')
+      .build();
+  }
+}
+
+export function getExecutor(): SfdxCommandletExecutor<{}> {
```

Comment:
I extracted the ternary into a function to make it easier for testing. I also think its a cleaner implementation and it mimics the pattern in other commands such as authWebLogin.